pkg/api/types.go

```diff
@@ -396,6 +398,24 @@ type NFSVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the lifetime of a pod
+type GlusterfsVolumeSource struct {
+	// Required: hosts are Glusterfs hosts address array
+	Hosts []string `json:"hosts"`
```

Comment:
I agree with @rootfs comments regarding DNS, as it may be tricky to have. Service with static Endpoints should be the way to go - so a service declaration example is needed.